### PR TITLE
Update test text to match spec.

### PR DESCRIPTION
**Pull request by BigBlueHat**
This includes minimal test changes. Some assert error messages were added and a
couple test fixtures where removed since they are no longer needed).

Details below:

- Update @context test to use ordered set language.
- Use a single MUST statement for IDs.
- Rename credential ID fixtures for easy findling.
- Remove duplicate fixture/test post consolidation.
- Update requirements text in Types section.
- Remove skipped SHOULD statement.
- Remove `proof` type tests.
- Add failure messages to some rejections.
- Add TODOs for Conformance section.
- Extract Verifiable Presentations section.
- Update existing VP test to use new text.
- Add pending test for VP id field.
- Add VP type pending test.
- Add pending tests for Enveloped VCs.
- Add VP holder claims pending tests.
- Create sections for subsections of VP tests.
- Move Data Schemas tests up from Advanced.
- Add some assert messages to help debugging.
- Begin updating Advanced section.
- Remove id tests in Refreshing.
- Unskip Refreshing tests.
- Add links to Refreshing section.
- Add links to termsOfUse tests.
- Add link and note to Evidence test.
- Remove skipped ZKP tests.
- Remove remaining two skipped tests.
- Add comments for possible future tests.
- Add .editorconfig.

### CODE REVIEW

File: tests/10-vcdm2.js [hunk 1]
```diff
@@ -201,22 +192,22 @@ describe('Types', function() {
...
-      it('Verifiable presentations MUST have a type property.',
-        async function() {
-          this.test.link = `https://w3c.github.io/vc-data-model/#types:~:text=Verifiable%20credentials%20and%20verifiable%20presentations%20MUST%20have%20a%20type%20property.`;
-          await assert.rejects(endpoints.verifyVp(require(
-            './input/presentation-no-type-fail.json')));
-        });
-      it('The value of the type property MUST be, or map to (through ' +
-        'interpretation of the @context property), one or more URLs.',
+      it('Verifiable credentials MUST contain a type property with an ' +
+        'associated value', async function() {
```
Comment (TallTed) [inline tests/10-vcdm2.js:196]:
```suggestion
        'associated value.', async function() {
```

File: tests/10-vcdm2.js [hunk 2]
```diff
@@ -201,22 +192,22 @@ describe('Types', function() {
...
+        this.test.link = `https://w3c.github.io/vc-data-model/#types:~:text=Verifiable%20credentials%20and%20verifiable%20presentations%20MUST%20contain%20a%20type%20property%20with%20an%20associated%20value.`;
+        await assert.rejects(
+          endpoints.issue(require('./input/credential-no-type-fail.json')));
+      });
+      it('Verifiable presentations MUST contain a type property with an ' +
+        'associated value', async function() {
```
Comment (TallTed) [inline tests/10-vcdm2.js:202]:
```suggestion
        'associated value.', async function() {
```

File: tests/10-vcdm2.js [hunk 3]
```diff
@@ -700,107 +722,237 @@ describe('Status', function() {
...
+  for(const [name, implementation] of match) {
+    const endpoints = new TestEndpoints({implementation, tag});
+
+    describe(name, function() {
+      beforeEach(addPerTestMetadata);
+
+      it('The @context property of the object MUST be present and include a ' +
+        'context, such as the base context for this specification, that ' +
+        'defines at least the id, type, and EnvelopedVerifiablePresentation ' +
+        'terms as defined by the base context provided by this specification',
```
Comment (TallTed) [inline tests/10-vcdm2.js:775]:
```suggestion
        'terms as defined by the base context provided by this specification.',
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Replacing apparently accidentally-dropped periods.

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# https://editorconfig.org
+
+root = true
```
Comment (aljones15) [inline .editorconfig:3]: which editor does this belong to? where is this needed? maybe this should be in the gitignore?

File: tests/10-vcdm2.js [hunk 5]
```diff
@@ -229,70 +220,58 @@ describe('Types', function() {
...
-        'MUST be interpreted as an unordered set."', async function() {
-        this.test.link = `https://w3c.github.io/vc-data-model/#types:~:text=If%20more%20than%20one%20URL%20is%20provided%2C%20the%20URLs%20MUST%20be%20interpreted%20as%20an%20unordered%20set.`;
-        //issue VC with multiple urls in type property
-        await endpoints.issue(require(
-          './input/credential-type-urls-order-1-ok.json'));
-        //issue another VC with same urls in a different order
-        await endpoints.issue(require(
-          './input/credential-type-urls-order-2-ok.json'));
-      });
+      it('If more than one value is provided, the order does not matter.',
```
Comment (aljones15) [inline tests/10-vcdm2.js:223]: ok so the way we've been handling this is with `(property name)` so in this case, "If more than one (type) value is provided, the order does not matter."

File: tests/10-vcdm2.js [hunk 6]
```diff
@@ -229,70 +220,58 @@ describe('Types', function() {
...
+        }
+      );
       // FIXME this needs to be expanded into at least 6 different tests
       // Verifiable Credential MUST have a type specified
       // Verifiable Presentation MUST have a type specified
-      // Proof MUST have a type specified.
       // credentialStatus MUST have a type specified.
       // termsOfUse MUST have a type specified.
       // evidence MUST have a type specified.
-      it('list: "objects that MUST have a type specified."',
```
Comment (aljones15) [inline tests/10-vcdm2.js:249]: thanks for removing this it has needed to be broken into other tests for awhile and good work here.

File: tests/10-vcdm2.js [hunk 7]
```diff
@@ -229,70 +220,58 @@ describe('Types', function() {
...
-          await assert.rejects(endpoints.issue(require(
-            './input/credential-status-missing-type-fail.json')));
-          await endpoints.issue(require(
-            './input/credential-termsofuse-ok.json'));
-          await assert.rejects(endpoints.issue(require(
-            './input/credential-termsofuse-missing-type-fail.json')));
-          await endpoints.issue(require('./input/credential-evidence-ok.json'));
-          await assert.rejects(endpoints.issue(require(
-            './input/credential-evidence-missing-type-fail.json')));
+      it('VerifiableCredential, VerifiablePresentation, credentialStatus, ' +
```
Comment (aljones15) [inline tests/10-vcdm2.js:240]: this is awesome and I really like it, good job.

File: tests/10-vcdm2.js [hunk 8]
```diff
@@ -229,70 +220,58 @@ describe('Types', function() {
...
+        'termsOfUse, and evidence are "objects that MUST have a type ' +
+        'specified."', async function() {
```
Comment (PatStLouis) [inline tests/10-vcdm2.js:242]: This list should also contain [`refreshService`](https://w3c.github.io/vc-data-model/#refreshing:~:text=Each%20refreshService%20value%20MUST%20specify%20its%20type.)

Comment (PatStLouis) [reply tests/10-vcdm2.js:242]: Same for [`credentialSchema`](https://w3c.github.io/vc-data-model/#data-schemas:~:text=Each%20credentialSchema%20MUST%20specify%20its%20type%20(for%20example%2C%20JsonSchema)

**Review by aljones15 — APPROVED** [review]
Approving as it adds the text, but not necessarily the test. good PR to move forward from and thanks for the work. Do what you want with that editorConfig file.

**Review by PatStLouis — APPROVED** [review]
Happy with the changes, I have a few more inconsistencies to add as well from my fork. Once this is merged in I'll pull these changes and open my PR

File: tests/10-vcdm2.js (hunk 6 above)
Comment (PatStLouis) [reply tests/10-vcdm2.js:249]: This is also covered in the advanced concepts section

File: tests/10-vcdm2.js (hunk 8 above)
Comment (TallTed) [reply tests/10-vcdm2.js:242]: I think this is what @PatStLouis suggests...
```suggestion
      it('VerifiableCredential, VerifiablePresentation, credentialStatus, ' +
        'termsOfUse, refreshService, credentialSchema, and evidence are ' +
        '"objects that MUST have a type specified."', async function() {
```

Comment (BigBlueHat) [reply tests/10-vcdm2.js:242]: Then that's a spec change that needs making, as the quote is from this section which references a table which lacks `refreshService` and `credentialSchema`: https://w3c.github.io/vc-data-model/#refreshing:~:text=objects%20that%20MUST%20have%20a%20type

I'll submit a PR there...and then we can update this test once that's merged into the spec.

Comment (BigBlueHat) [reply tests/10-vcdm2.js:242]: @TallTed @PatStLouis submitted https://github.com/w3c/vc-data-model/pull/1530

File: .editorconfig (hunk 4 above)
Comment (BigBlueHat) [reply .editorconfig:3]: No. It doesn't ignore anything. It enforces other things that eslint overlooks. Read about it at https://editorconfig.org it works in loads of editors. We've also been gradually adding them to repos to help keep things tidy/consistent.

File: tests/10-vcdm2.js (hunk 5 above)
Comment (BigBlueHat) [reply tests/10-vcdm2.js]: Fixed in [41f2a0f](https://github.com/w3c/vc-data-model-2.0-test-suite/pull/81/commits/41f2a0f4efff0212e1e19e5fe6526cd1d023256a)

**Review by PatStLouis — APPROVED** [review]
Looks good

**Review by aljones15 — APPROVED** [review]
approving again provided this PR is checked to ensure commits in it are not duplicates of existing commits in main (might need a rebase from main in other words).

**Pull request merged**